pymatgen/transformations/advanced_transformations.py

```diff
@@ -1360,3 +1361,213 @@ def inverse(self):
     @property
     def is_one_to_many(self):
         return False
+
+class CubicSupercellTransformation:
+    """
+    A transformation that aims to generate a nearly cubic supercell structure from a structure.
+    """
+
+    def __init__(self, structure, min_atoms=-np.Inf, max_atoms=np.Inf, num_nn_dists=5):
+        """
+        Returns the transformation matrix of a Pymatgen structure into a suitable supercell for CSLD
+
+        Args:
+            structure (Structure): input structure.
+            max_atoms (int): maximum number of atoms allowed in the supercell
+            min_atoms (int): minimum number of atoms allowed in the supercell
+            num_nn_dists (int): number of multiples of atomic nearest neighbor distances to
+                force all directions of the supercell to be at least as large
+        Returns:
+            Supercell structure (Structure)
+        """
+        self.structure = structure
+        self.min_atoms = min_atoms
+        self.max_atoms = max_atoms
+        self.num_nn_cutoff = num_nn_dists
+
+        self.lat_vecs = self.structure.lattice.matrix
+        bond_matrix = self.structure.distance_matrix
+        np.fill_diagonal(bond_matrix, np.Inf)
+        self.nn_dist = np.amin(bond_matrix)
+
+        self.smallest_dim = None # smallest dimension of the resulting supercell
+
+    def round_away_from_zero(self, x):
```

Comment:
This function should be private.

Also, what is it actually doing? Can you add some docs.